presto-hive/src/main/java/com/facebook/presto/hive/statistics/ParquetQuickStatsBuilder.java

```diff
@@ -0,0 +1,344 @@
+import org.apache.parquet.column.statistics.FloatStatistics;
+import org.apache.parquet.column.statistics.IntStatistics;
+import org.apache.parquet.column.statistics.LongStatistics;
+import org.apache.parquet.column.statistics.Statistics;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.LogicalTypeAnnotation.DateLogicalTypeAnnotation;
+import org.apache.parquet.schema.LogicalTypeAnnotation.LogicalTypeAnnotationVisitor;
+import org.apache.parquet.schema.LogicalTypeAnnotation.TimeLogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+
+import java.time.LocalDate;
+import java.time.chrono.ChronoLocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.hive.CacheQuota.NO_CACHE_CONSTRAINTS;
+import static com.facebook.presto.hive.HiveCommonSessionProperties.getReadNullMaskedParquetEncryptedValue;
+import static com.facebook.presto.hive.HivePartition.UNPARTITIONED_ID;
+import static com.facebook.presto.hive.parquet.HdfsParquetDataSource.buildHdfsParquetDataSource;
+import static com.facebook.presto.hive.parquet.ParquetPageSourceFactory.PARQUET_SERDE_CLASS_NAMES;
+import static com.facebook.presto.hive.parquet.ParquetPageSourceFactory.createDecryptor;
+import static com.facebook.presto.parquet.cache.MetadataReader.readFooter;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class ParquetQuickStatsBuilder
+        implements QuickStatsBuilder
+{
+    public static final Logger log = Logger.get(ParquetQuickStatsBuilder.class);
+    private static final ExecutorService footerFetchExecutor = Executors.newCachedThreadPool();
+    private final HdfsEnvironment hdfsEnvironment;
+    private final FileFormatDataSourceStats stats;
+    private final long footerFetchTimeoutMillis;
+
+    public ParquetQuickStatsBuilder(FileFormatDataSourceStats stats, HdfsEnvironment hdfsEnvironment, HiveClientConfig hiveClientConfig)
+    {
+        this.stats = stats;
+        this.hdfsEnvironment = hdfsEnvironment;
+        this.footerFetchTimeoutMillis = hiveClientConfig.getParquetQuickStatsFooterFetchTimeout().roundTo(MILLISECONDS);
+    }
+
+    private static void processColumnMetadata(ParquetMetadata parquetMetadata, Map<ColumnPath, ColumnQuickStats<?>> rolledUpColStats)
```

Comment:
nit: Move private methods to after the public ones

Reply:
Can you share your codestyle rules for IntelliJ ?
Mine have constructor above static methods
![image](https://github.com/prestodb/presto/assets/1797669/4148e6a6-d7f1-4ae1-840f-788f4b74d963)